_includes/v23.1/sql/select-for-update-limitations.md

```diff
@@ -0,0 +1,5 @@
+Locks acquired using {% if page.name == "select-for-update.md" %} `SELECT ... FOR UPDATE` {% else %} [`SELECT ... FOR UPDATE`](select-for-update.html) {% endif %} are lost on [lease transfers](architecture/replication-layer.html#epoch-based-leases-table-data) and [range splits and merges](architecture/distribution-layer.html#range-merges). This happens because currently `SELECT ... FOR UPDATE` is implemented using [unreplicated locks](architecture/transaction-layer.html#unreplicated-locks).
+
+In most cases, this will cause the transaction that acquired the locks to fail to commit. For example, this behavior can lead to [transaction retry errors with code `40001` and the `restart transaction` error message](common-errors.html#restart-transaction).
```

Comment:
This could benefit from being even more explicit about the consequences of the locks being dropped instead of leaving the "in other cases" scenario up for interpretation. Notably, in all cases, serializable isolation is not violated.

Reply:
I took a look at a few of the issues like https://github.com/cockroachdb/cockroach/issues/83571 and https://github.com/cockroachdb/cockroach/issues/75456 and updated this to be more explicit and list some explicit behaviors.  

PTAL and let me know if they are accurate and/or if we need to list more behaviors